File: packages/es-markdown-core/src/model/directive-label.ts
import type { MarkupBase } from "./base.js";
import type { Markup } from "./markup.js";

/**
 * A directive's bracketed label. Its scope spans the brackets, so a label
 * written empty is still a place in the source.
 */
export interface DirectiveLabel extends MarkupBase<"directiveLabel"> {
    readonly content: readonly Markup[];
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/DirectiveLabel.kt
package com.nouprax.markdown.core

/**
 * A directive's bracketed label. It is Markup owned by the directive's label
 * field, not directive content. Its scope spans the brackets, so a label
 * written empty is still a place in the source.
 */
public class DirectiveLabel internal constructor(
    public val content: kotlin.collections.List<Markup>,
    override val scope: Scope,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visitDirectiveLabel(this)
}
